utils/python/CIME/build.py

```diff
@@ -563,3 +563,55 @@ def _build_model_thread(config_dir, caseroot, bldroot, compspec, file_build,
 
     for mod_file in glob.glob(os.path.join(objdir, "*_[Cc][Oo][Mm][Pp]_*.mod")):
         shutil.copy(mod_file, incroot)
+
+###############################################################################
+def clean(case, cleanlist):
+###############################################################################
+    debug           = case.get_value("DEBUG")
+    use_esmf_lib    = case.get_value("USE_ESMF_LIB")
+    build_threaded  = case.get_value("BUILD_THREADED")
+    clm_config_opts = case.get_value("CLM_CONFIG_OPTS")
+    testcase        = case.get_value("TESTCASE")
+    gmake           = case.get_value("GMAKE")
+    caseroot        = case.get_value("CASEROOT")
+    casetools       = case.get_value("CASETOOLS")
+
+    os.environ["DEBUG"]           = stringify_bool(debug)
+    os.environ["USE_ESMF_LIB"]    = stringify_bool(use_esmf_lib)
+    os.environ["BUILD_THREADED"]  = stringify_bool(build_threaded)
+    os.environ["CASEROOT"]        = case.get_value("CASEROOT")
+    os.environ["COMP_INTERFACE"]  = case.get_value("COMP_INTERFACE")
+    os.environ["PIO_VERSION"]     = str(case.get_value("PIO_VERSION"))
+    os.environ["CLM_CONFIG_OPTS"] = clm_config_opts  if clm_config_opts is not None else ""
+
+    if testcase is not None:
+        if clm_config_opts is not None:
+            if "clm4_0" in clm_config_opts:
+                cleanlist.remove('cleanlnd')
+
+    cmd = gmake + " -f " + casetools + "/Makefile"
+    for item in cleanlist:
+        cmd = cmd + " clean" + item
+    logger.info("calling %s "%(cmd))
+    run_cmd(cmd)
+
+    # unlink Locked files directory
+    file = os.path.join(caseroot,"LockedFiles/env_build.xml")
+    if os.path.isfile(file):
+        os.unlink(file)
+
+    # reset following values in xml files
+    case.set_value("SMP_BUILD",str(0))
+    case.set_value("NINST_BUILD",str(0))
+    case.set_value("BUILD_STATUS",str(0))
+    case.set_value("BUILD_COMPLETE","FALSE")
+    case.flush()
+
+    # append call of to CaseStatus 
+    cmd = "date +\"%Y-%m-%d %H:%M:%S\""
```

Comment:
Yes, or use the method out of utils.py that provides the standard CIME timestamp string.